presto-array/src/main/java/com/facebook/presto/array/Arrays.java

```diff
@@ -52,4 +60,33 @@ private Arrays() {}
 
         return buffer;
     }
+
+    public static byte[] ensureCapacity(byte[] buffer, int capacity, float expansionFactor, boolean preserveValues)
```

Comment:
There're two notions here: 
capacity: below which the array will be resized. This is the minimum required size.
newCapacity: the new capacity this array will be grown to. 

We can pass in the newCapacity as capacity * expansionFactor, but we still need to pass it in as an argument. In this sense it's more straightforward to pass in the expansionFactor instead of newCapacity.